sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/identity/providers/mysql/AzureIdentityMysqlAuthenticationPlugin.java

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.service.implementation.identity.providers.mysql;// Copyright (c) Microsoft Corporation. All rights reserved.
+
+import com.azure.spring.cloud.service.implementation.identity.api.AuthProperty;
+import com.azure.spring.cloud.service.implementation.identity.api.AzureAuthenticationTemplate;
+import com.mysql.cj.callback.MysqlCallbackHandler;
+import com.mysql.cj.log.LogFactory;
+import com.mysql.cj.protocol.AuthenticationPlugin;
+import com.mysql.cj.protocol.Protocol;
+import com.mysql.cj.protocol.a.NativeConstants;
+import com.mysql.cj.protocol.a.NativePacketPayload;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * The Authentication plugin that enables Azure AD managed identity support.
+ */
+public class AzureIdentityMysqlAuthenticationPlugin extends AzureAuthenticationTemplate implements AuthenticationPlugin<NativePacketPayload> {
+    // TODO log type
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureIdentityMysqlAuthenticationPlugin.class);
+
+    private static String PLUGIN_NAME = "mysql_clear_password";
+
+    /**
+     * Stores the callback handler.
+     */
+    private MysqlCallbackHandler callbackHandler;
+
+    /**
+     * Stores the protocol.SharedTokenCacheCredential
+     */
+    private Protocol<NativePacketPayload> protocol;
+
+    private String sourceOfAuthData;
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public String getProtocolPluginName() {
+        return PLUGIN_NAME;
+    }
+
+    @Override
+    public void init(Protocol<NativePacketPayload> protocol) {
+        this.protocol = protocol;
+        Properties properties = protocol.getPropertySet().exposeAsProperties();
+        AuthProperty.SCOPES.setProperty(properties, "https://ossrdbms-aad.database.windows.net/.default");
```

Comment:
A constant for "https://ossrdbms-aad.database.windows.net/.default"?

Reply:
ok